authors/apps/articles/views.py

```diff
@@ -34,66 +40,203 @@ def get(self, request, **kwargs):
+            response = {"success": "Article successfully updated!",
+                        "data": serializer.data}
             return Response(response, status.HTTP_200_OK)
         return Response({
-                 "message": "Please use proper request body!",
-                 "status":status.HTTP_400_BAD_REQUEST
-            }, status.HTTP_400_BAD_REQUEST)
+            "message": "Please use proper request body!",
+            "status": status.HTTP_400_BAD_REQUEST
+        }, status.HTTP_400_BAD_REQUEST)
 
     def delete(self, request, **kwargs):
         is_authenticated(request)
         article = Article.objects.filter(slug=kwargs.get('slug')).first()
         if not article:
             return Response({
-                 "message": "Article does not exist"
+                "message": "Article does not exist"
             }, status.HTTP_404_NOT_FOUND)
-        
+
         serializer = self.serializer_class(
-            article, 
+            article,
             partial=True
         )
         article.delete()
-        response = {"success": "Article successfully deleted!", "article": serializer.data}
+        response = {"success": "Article successfully deleted!",
+                    "article": serializer.data}
         return Response(response, status.HTTP_200_OK)
 
 def is_authenticated(request):
     user = request.user.id
-    
+
     if user == None:
         raise PermissionDenied({
-            "message":"Please Login to proceed",
-            "status":status.HTTP_403_FORBIDDEN
+            "message": "Please Login to proceed",
+            "status": status.HTTP_403_FORBIDDEN
         })
+
+class FavoriteUnfavoriteAnArticle(CreateAPIView, DestroyAPIView):
+    """If the user feels satisfied with the article, he can favourite it """
+
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    queryset = FavoriteAnArticle.objects.all()
+    serializer_class = FavoriteAnArticleSerializer
+    article_serializer_class = ArticleSerializer
+
+    def check_article_exists(self, **kwargs):
+        """
+        Check whether the article with specified slug exists
+        """
+        article = get_object_or_404(Article, slug=kwargs.get("slug"))
+        return article
+
+    @swagger_auto_schema(
+        operation_description='Favorite an Article.',
+        operation_id='favorite an article',
+        responses={200: serializer_class(many=False), 400: 'BAD REQUEST'},
+    )
+    def post(self, request, **kwargs):
+        """
+        Favorite an article with specified slug
+        """
+        article = self.check_article_exists(**kwargs)
+        favorite = FavoriteAnArticle.objects.filter(
+            favorited_by=request.user, article=article
+        )
+        if not favorite:
+            serializer = self.serializer_class(data={"is_favorited": True})
+            serializer.is_valid(raise_exception=True)
+            serializer.save(article=article, favorited_by=request.user)
```

Comment:
@alexxsanya , Please work on these codeclimate issues

Reply:
@cdvx okay.

Reply:
@cdvx this has been fixed